charms/istio-gateway/src/charm.py

```diff
@@ -162,5 +220,68 @@ def __init__(self, msg, status_type=StatusBase):
         self.status = status_type(self.msg)
 
 
+def valid_annotations(annotations: Dict[str, str]) -> bool:
+    """Return True if the annotations pass a series of checks, False otherwise.
+
+    Validate the annotations against the following checks:
+    * Length
+    * Annotation syntax
+    * Reserved prefixes
+
+    Args:
+      annotations (Dict[str,str]): dictionary with annotations
+    """
+    for key, value in annotations.items():
+        return validate_annotation_key(key) and validate_annotation_value(value)
+    return True
+
+
+def is_qualified_name(value: str) -> bool:
+    """Check if a value is a valid Kubernetes qualified name."""
+    parts = value.split("/")
+    if len(parts) > 2:
+        return False  # Invalid if more than one '/'
+
+    if len(parts) == 2:  # If prefixed
```

Comment:
suggestion: maybe we can move constants to the top of file

Reply:
See https://github.com/canonical/istio-operators/pull/611#discussion_r2077641026